src/h5web/visualizations/heatmap/utils.ts

```diff
@@ -17,3 +19,28 @@ export function generateCSSLinearGradient(
 
   return `linear-gradient(to ${direction},${gradientColors})`;
 }
+
+export function getColorScaleDomain(
+  scaleType: ScaleType,
+  values: number[],
+  dataDomain?: Domain,
+  customDomain?: Domain
+): Domain | undefined {
+  const domain = customDomain || dataDomain;
+  if (!domain) {
+    return undefined;
+  }
+
+  // If the scale is not log or if the domain does not cross zero, the domain does not need to be adjusted
+  if (scaleType !== ScaleType.Log || domain[0] * domain[1] > 0) {
+    return domain;
+  }
+
+  const supportedDomain = findDomain(values.filter((x) => x > 0));
+  if (supportedDomain && customDomain) {
+    // Clamp custom domain minimum to the first positive value
+    return [supportedDomain[0], customDomain[1]] as Domain;
+  }
+
+  return supportedDomain;
+}
```

Comment:
This is the main point of the MR: adding a function to compute once and for all components needing it, the domain to be used when mapping to colours.